src/v/cluster/types.h

```diff
@@ -3908,6 +3908,26 @@ struct metrics_reporter_cluster_info
     auto serde_fields() { return std::tie(uuid, creation_timestamp); }
 };
 
+struct controller_last_applied_request
+  : serde::envelope<
+      controller_last_applied_request,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+};
+
+struct controller_last_applied_reply
+  : serde::envelope<
+      controller_last_applied_reply,
+      serde::version<0>,
+      serde::compat_version<0>> {
+    using rpc_adl_exempt = std::true_type;
+    model::offset last_applied;
+    errc result;
+
+    auto serde_fields() { return std::tie(last_applied); }
```

Comment:
I think you need `errc` here too.

Are there roundtrip tests or serde backwards compat tests we can add for this?

Reply:
nice catch